cmake/modules/autopas_antlr4cpp.cmake

```diff
@@ -1,101 +1,119 @@
-        CMAKE_ARGS      -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=${UTFCPP_DIR}/install -DUTF8_TESTS=off -DUTF8_SAMPLES=off
-    )
-else()
-    message(STATUS "utf8cpp found - using system version")
-    # add a dummy target so the dependency in antlr4cpp_bundled is fulfilled if utf8cpp was found on the system
-    add_custom_target(utf8cpp_bundled)
-endif ()
-
-# install prefix for antlr
-set(antlr4cpp_prefix ${CMAKE_CURRENT_BINARY_DIR}/_deps/antlr4cppPrefix)
-# location where antlr will install its static library
-set(staticLibInstallLocation ${antlr4cpp_prefix}/install/lib/libantlr4-runtime.a)
-
-ExternalProject_ADD(
-        antlr4cpp_bundled
-        PREFIX           ${antlr4cpp_prefix}
-        URL              ${PROJECT_SOURCE_DIR}/libs/antlr4-cpp-runtime-4.9.3-source.zip
-        URL_HASH         MD5=eafa4fef583e12e963062882773461be
-        BUILD_BYPRODUCTS ${staticLibInstallLocation}
-        # pass PKG_CONFIG_PATH as a environment variable to cmake so find_package() in antlr4cpp's CMakeLists.txt can find uuid-dev if using the bundled version
-        CMAKE_COMMAND    ${CMAKE_COMMAND} -E env PKG_CONFIG_PATH=${LIBUUID_PKGCONFIG_DIR} ${CMAKE_COMMAND}
-        BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/antlr4cpp/install/lib/libantlr4-runtime.a
-        # point antlr4cpp to utf8cpp install dir
-        CMAKE_ARGS       -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=${antlr4cpp_prefix}/install -DCMAKE_CXX_FLAGS=-w -DCMAKE_PREFIX_PATH=${UTFCPP_DIR}/install
-        # make sure UUID and UTF8CPP is installed before antlr is installed
-        DEPENDS          uuid_bundled utf8cpp_bundled
-        # Patch away the shared library target
-        PATCH_COMMAND    sed --in-place -e /install.*shared/,+2d               runtime/CMakeLists.txt  &&
-                         sed --in-place -e /set_target_properties.*shared/,+9d runtime/CMakeLists.txt  &&
-                         sed --in-place -e /shared/d                           runtime/CMakeLists.txt
-)
-
-# create dummy target that contains all information to easily link against
-add_library(antlr4cpp
-        STATIC
-        IMPORTED
-        GLOBAL
+if (AUTOPAS_ENABLE_RULES_BASED_TUNING)
+    message(STATUS "Rules-Based Tuning Enabled")
+
+
+    message(STATUS "antlr4cpp - using bundled version")
+
+    include(ExternalProject)
+
+    # check if uuid-dev is installed on the system, since this is a dependency of antlr4cpp
+    find_package(PkgConfig)
+    pkg_check_modules(UUID QUIET uuid)
+
+    # if uuid-dev was not found on system we install it locally
+    if (NOT UUID_FOUND)
+        message(STATUS "UUID not found - using bundled version")
+        message(WARNING "Compilation of the bundled UUID is known to fail. If it does either:
+                        1) Install UUID on your machine yourself.
+                        2) Disable rules-based tuning via the CMake option -DAUTOPAS_ENABLE_RULES_BASED_TUNING=OFF.
+                        3) (Untested!) Install automake 1.13.")
```

Comment:
I agree, but - if I am correct in my understanding of the `uuid` problem - the bundled `uuid` will always fail to compile unless the machine has automake 1.13 (which is usually not the case). 

As such, this is less of a blanket warning and more of a "foretelling" of what will happen. Hence, I would see this less as blanket spam for 95% of cases and more as useful information except for 5% of cases. (Numbers are arbitrary but you get the point).

This can save the user time in debugging the error and eliminate doubt that the error was caused by them making a mistake. This is especially important for application scientists with less experience. 

1 is obvious, given that the user already understands that the bundled version is problematic. But I see no reason not to spell it out for novice HPC users and application scientists, given the two points above.

2 is not obvious. You cannot expect novice users of AutoPas to know every CMake option and what dependencies they stop from needing to be compiled when disabled, especially as the name of the option does not make it clear that this will stop `uuid` from being needed.

3 is not a random guess, as the error caused when compiling `uuid` on the ARM cluster is

~~~bash
WARNING: 'automake-1.13' is missing on your system.
~~~

which implies that installing it would fix the system. I just have not tested it because I didn't care about finding a solution to compile rules-based tuning on the ARM cluster (at least for the moment). I am happy to remove this if you still think it is not needed.

Reply:
I agree. Fixed in `5044d8b5`. I added some information to `Building.md` to provide a little information about the matter.